src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1992,50 +1972,41 @@ private bool IsPossibleTypeParameterConstraint()
 
         private TypeParameterConstraintSyntax ParseTypeParameterConstraint()
         {
-            SyntaxToken questionToken = null;
-            var syntaxKind = this.CurrentToken.Kind;
-
             switch (this.CurrentToken.Kind)
             {
                 case SyntaxKind.NewKeyword:
-                    var newToken = this.EatToken();
-                    var open = this.EatToken(SyntaxKind.OpenParenToken);
-                    var close = this.EatToken(SyntaxKind.CloseParenToken);
-                    return _syntaxFactory.ConstructorConstraint(newToken, open, close);
+                    return _syntaxFactory.ConstructorConstraint(
+                        this.EatToken(),
```

Comment:
Maybe name the argument since it's no longer clear this is `newToken`?

```suggestion
                        newKeyword: this.EatToken(),
```

And similarly below where parameterless `this.EatToken()` is used.